### PR TITLE
Mobile view calendar improvements

### LINKED ISSUE #14997: Calendar card: Scale buttons dynamically so they are actually shown on narrow screens/columns
### Checklist

- [X] I have updated to the latest available Home Assistant version.
- [X] I have cleared the cache of my browser.
- [X] I have tried a different browser to see if it is related to my browser.

### Describe the issue you are experiencing

When the screen/column is broad enough, the [calendar card](https://www.home-assistant.io/dashboards/calendar/) looks just fine:
grafik

Once screen/column is quite narrow, calendar card's elements are cut, especially the list/day/month filter buttons (2nd issue: automatically the most right element - list view - is activated):

grafik

### Describe the behavior you expected

Align buttons in a way
a) they are all shown at all and
b) not the list view is being activated at all, but the view set using `initial_view` is kept

### Steps to reproduce the issue

1. Add calendar card
2. Resize window so it gets narrow / alternatively use a mobile device with smaller screen

### What version of Home Assistant Core has the issue?

core-2022.10.5

### What was the last working version of Home Assistant Core?

_No response_

### In which browser are you experiencing the issue with?

Several - Firefox on Windows, HA app on iOS and Android, ...

### Which operating system are you using to run this browser?

Several - see question above

### State of relevant entities

_No response_

### Problem-relevant frontend configuration

_No response_

### Javascript errors shown in your browser console/inspector

_No response_

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by silamon**
## Proposed change
* Allow the buttons to wrap on a very narrow mobile view 
* Allow more views to be used in a calendar card. (Align with the calendar view.)

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (thank you!)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [x] Code quality improvements to existing code or addition of tests

## Example configuration

```yaml

```

## Additional information

- This PR fixes or closes issue: fixes #14997
- This PR is related to issue or discussion:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] There is no commented out code in this PR.
- [ ] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/home-assistant/home-assistant.io

### CODE REVIEW

**Comment by bramkragten** [comment]
Can you share a screenshot how this looks now?

**Comment by silamon** [comment]
Monthly recheck: This still needs a review.

File: src/panels/calendar/ha-panel-calendar.ts [hunk 1]
```diff
@@ -187,7 +187,7 @@ class PanelCalendar extends LitElement {
           .events=${this._events}
           .calendars=${this._calendars}
           .narrow=${this.narrow}
-          .initialView=${this.narrow ? "listWeek" : "dayGridMonth"}
+          initialView="dayGridMonth"
```
Comment (bramkragten) [inline src/panels/calendar/ha-panel-calendar.ts:190]: I don't think `dayGridMonth` works well on small screens, that's why I didn't make it the default there

**Comment by bramkragten** [comment]
Thanks, this card (and the calendar panel) is currently being re-designed by the product team. But I think we can merge this for now. See my comment.

**Comment by frenck** [comment]
@silamon Are you on Discord? If so, could you reach out to me on it? (I'm also @frenck on that platform).

**Review by bramkragten — APPROVED** [review]

**Pull request merged**